### ISSUE TITLE
twisted.vfs.test.test_vfs.OSVFSTest.setUp does not close the files it opens

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2974]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2974|
|Type|defect|
|Created|2008-01-05 15:02:13Z|
|Branch|https://github.com/twisted/twisted/tree/osvfstest-write-2974|

The setUp method relies on the garbage collector to collect and therefore flush the file it writes to.  If this does not happen before certain of the tests run, they will fail since the bytes expected will not be available in the file.

<details><summary>Searchable metadata</summary>

```
trac-id__2974 2974
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_osvfstest_write_2974 branches/osvfstest-write-2974
branch_author__ 
status__closed closed
resolution__fixed fixed
component__vfs vfs
keywords__ 
time__1199545333000000 1199545333000000
changetime__1221064385000000 1221064385000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [24772]) Branching to 'osvfstest-write-2974'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [24773]) Use FilePath.setContent instead of file.write

refs [#2974](https://github.com/twisted/twisted/issues/2974)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Should be fixed

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Fine, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [24778]) Merge osvfstest-write-2974

Author: exarkun
Reviewer: therve
Fixes: [#2974](https://github.com/twisted/twisted/issues/2974)

Ensure that the test file created in the VFS unit test setup
fixture is flushed and closed before any of the tests try
to use its contents.  This removes the reliance on reference
counting to free the file object and flush the buffer in
order for the tests to pass.
